scripts/west_commands/runners/stm32flash.py

```diff
@@ -35,47 +42,80 @@ def capabilities(cls):
     def do_add_parser(cls, parser):
 
         # required argument(s)
+        # none for now
+
         # optional argument(s)
         parser.add_argument('--device', default=default_device, required=False,
-                            help='serial port to flash, default ' + default_device)
-
-        parser.add_argument('--start-address', default='0x0',required=False,
-                            help='specify start address for write operation, default 0x0')
+                            help='serial port to flash, default \'' + default_device + '\'')
 
         parser.add_argument('--baud-rate', default='57600', required=False,
                             choices=['1200', '1800', '2400', '4800', '9600', '19200',
                             '38400', '57600', '115200', '230400', '256000', '460800',
                             '500000', '576000', '921600', '1000000', '1500000', '2000000'],
-                            help='serial baud rate, default 57600')
+                            help='serial baud rate, default \'57600\'')
+
+        parser.add_argument('--force-binary-parser', required=False, action='store_true',
+                            help='force binary parser')
+
+        parser.add_argument('--get-info', required=False, action='store_true',
+                            help='get device information')
+
+        parser.add_argument('--start-address', default='0',required=False,
+                            help='specify start address for write operation, default \'0\'')
+
+        parser.add_argument('--execution-addr', default=None,required=False,
+                            help='start execution at specified address, default: \'auto\' \
+                                  equals same as start address')
+
+        parser.add_argument('--serial-mode', default='8e1',required=False,
+                            help='serial port mode, default \'8e1\'')
 
-        parser.add_argument('--firmware', default='zephyr.bin', required=False,
```

Comment:
Why was this added and removed?

It's OK (and in fact very welcome!) if you want to structure your PR so that the commits build on themselves, but it makes it harder for me to review if things come and go like this.

Would you mind squashing things together to present the final results (possibly building it up from small pieces, though for something self-contained like a new runner, it's no big deal to just present the entire Python script in a single commit) instead of adding/taking away functionality? It would save me some time reviewing and I'd really appreciate it. I (and many other Zephyr reviewers you will encounter) read PR commits in order, so this is a big help.

Thanks!

Reply:
Hi @mbolivar,

A question about the rebasing: I can squash them into one commit be rebasing it to <insert some branch> locally but I'm not able to push it to github. Although the remote and local runner-stm32flash branches are identical before the rebase.
Afterwards git complains that the branches diverged - which of course is true.

How do I handle that? Are you ok if I force push the runner-stm32flash branch? Or do I create a new PR?